### PR TITLE
NoTileLeftBehind should return 404s

### PR DESCRIPTION
**Pull request by mojodna**
`NoTileLeftBehind` is intended to bypass caches. It does for local caches, but it inherits the default rendering behavior, which is to return a 200 HTTP response. This means that empty tiles (not intended to be cached) are treated as cacheable by downstream requestors.

### CODE REVIEW

**Comment by rburhum** [comment]
Makes sense

**Comment by mojodna** [comment]
I just realized a slight problem with this: `NoTileLeftBehind` includes a `tile` attribute which is intended to pass through whatever value is intended to be displayed.  This behavior is potentially useful (Proxy-provided tiles which have `max-age=0` or `no-cache` Cache-Control headers).

After #236 (which was the fundamental problem I was trying to address with this), it may make sense to revert to the original behavior (return `200`s) **and** add `Cache-Control: no-cache` to discourage downstream caches from treating it as a complete response.

In other words, bounds checks (hard errors) should be treated differently from intermittent errors (which may return content, even if it's not intended to last), the status quo prior to this PR.

Thoughts?

**Comment by juliomalegria** [comment]
I think the answer for this one depends on what we decide to do in #236.

**Comment by mojodna** [comment]
After letting this soak in for a while, I'm really of the opinion that this PR is wrong.  #238 reverts it, #239 adds `Cache-Control: no-cache` headers to match the intent of `NoTileLeftBehind`.

#236 can remain as a separate question.  I'm also hoping to submit some patches to the `Proxy` provider to respect the intent of upstream providers.
